### PR TITLE
Slapping now hurts if you slap someone with a helmet

### PR DESCRIPTION
**Pull request by FreeStylaLT**
What the title says, basically, slapping someone with a helmet that has
HEADCOVERSMOUTH flag will give the slapper brute damage, because that
would hurt. Also changes the slap message type.
12090a0b7f

### CODE REVIEW

**Comment by Iamgoofball** [comment]
walk up to someone
say "*slap
ctrl + up arrow on repeat till dead

:+1:

File: code/modules/mob/living/carbon/human/emote.dm [hunk 1]
```diff
@@ -690,10 +690,16 @@
...
+					if(ishuman(H) && H.head && (H.head.flags & HEADCOVERSMOUTH))
+						message = "<span class='danger'>[src] tries to slap [H] across the face, but hits their [H.head] instead! Ouch!"
+						playsound(src.loc, 'sound/effects/snap.ogg', 50, 1)
+						src.adjustBruteLoss(4)
+					else
+						message = "<span class = 'danger'>[src] slaps [M] across the face. Ouch!"
+						playsound(src.loc, 'sound/effects/snap.ogg', 50, 1)
 				else
-					message = "\red <B>[src]</B> slaps \himself!"
+					message = "<span class = 'danger'>[src] slaps themselves!"
```
Comment (tigercat2000) [inline code/modules/mob/living/carbon/human/emote.dm]: >spaces between class, =, and 'danger'

wai u do dis hawke

Comment (Iamgoofball) [inline code/modules/mob/living/carbon/human/emote.dm]: go away, this is readable

Comment (TheDZD) [inline code/modules/mob/living/carbon/human/emote.dm]: Forgot the </span> at the end of the message though.

Comment (tigercat2000) [inline code/modules/mob/living/carbon/human/emote.dm]: it's not consistent with anything! Not even the span class three lines above it!

File: code/modules/mob/living/carbon/human/emote.dm [hunk 2]
```diff
@@ -690,10 +690,16 @@
 							M = A
 							break
 				if (M)
-					message = "\red <B>[src]</B> slaps [M] across the face. Ouch!"
-					playsound(src.loc, 'sound/effects/snap.ogg', 50, 1)
+					var/mob/living/carbon/human/H = M
+					if(ishuman(H) && H.head && (H.head.flags & HEADCOVERSMOUTH))
+						message = "<span class='danger'>[src] tries to slap [H] across the face, but hits their [H.head] instead! Ouch!"
```
Comment (TheDZD) [inline code/modules/mob/living/carbon/human/emote.dm]: </ span> is missing from the end of the message. Omit the space inside of it though, that is included here so it'll show up on Github.

File: code/modules/mob/living/carbon/human/emote.dm [hunk 3]
```diff
@@ -690,10 +690,16 @@
...
 				if (M)
-					message = "\red <B>[src]</B> slaps [M] across the face. Ouch!"
-					playsound(src.loc, 'sound/effects/snap.ogg', 50, 1)
+					var/mob/living/carbon/human/H = M
+					if(ishuman(H) && H.head && (H.head.flags & HEADCOVERSMOUTH))
+						message = "<span class='danger'>[src] tries to slap [H] across the face, but hits their [H.head] instead! Ouch!"
+						playsound(src.loc, 'sound/effects/snap.ogg', 50, 1)
+						src.adjustBruteLoss(4)
+					else
+						message = "<span class = 'danger'>[src] slaps [M] across the face. Ouch!"
```
Comment (TheDZD) [inline code/modules/mob/living/carbon/human/emote.dm]: See previous comment.

**Comment by Fox-McCloud** [comment]
Not a big fan of this, to be honest--one could easily still slap someone upside the head or on the face even if they're wearing a helmet--and do so in such a way that you're not going to hurt yourself (people bwoink each other on the head, despite wearing hardhats, all the time). 

Also, no reason to have "themselves" \himself changes it depending the mob's gender (including "itself").

File: code/modules/mob/living/carbon/human/emote.dm [hunk 4]
```diff
@@ -690,10 +690,16 @@
...
+					var/mob/living/carbon/human/H = M
+					if(ishuman(H) && H.head && (H.head.flags & HEADCOVERSMOUTH))
+						message = "<span class='danger'>[src] tries to slap [H] across the face, but hits their [H.head] instead! Ouch!</span>"
+						playsound(src.loc, 'sound/effects/snap.ogg', 50, 1)
+						src.adjustBruteLoss(4)
+					else
+						message = "<span class='danger'>[src] slaps [M] across the face. Ouch!</span>"
+						playsound(src.loc, 'sound/effects/snap.ogg', 50, 1)
 				else
-					message = "\red <B>[src]</B> slaps \himself!"
```
Comment (Iamgoofball) [inline code/modules/mob/living/carbon/human/emote.dm]: Revert this (but keep the span class part)

\himself works just fine here

**Comment by ZomgPonies** [comment]
I have no particular opinion on this

**Comment by Fox-McCloud** [comment]
Ahahah, same for @TheDZD. I'd like to add that I'm not going to make an issue of this NOT being merged in, I just personally favor slapping being universal for roleplay reasons, s'all. Would really like more feedback on this so we can gauge if this needs to go in or not.

**Comment by tigercat2000** [comment]
I'm firmly meh on this.

**Comment by Earthdivine** [comment]
I don't see why not , your slapping a helmet, of course it's gunna make a thunk and hurt. And if it's an issue with Biohoods, remember their film is so hard Facehuggers kill themselves on it.

**Comment by Robertha89** [comment]
While it seems like a neat idea, it's stupid due to the simple fact that slapping other people does not hurt them, you can only slap yourself to death, and slapping others is just used for RP.

This commit is basically punishing RP, so a :100: x :-1: for me.

We should encourage rp, not punish it.

**Comment by FreeStylaLT** [comment]
@Robertha89 I'm not sure you got it right, because the *slapper* is the one that gets hurt if he slaps someone with a helmet, not the slapee (the person with the helmet). I don't see how this is punishing RP by having the person who slaps someone with a sec hardsuit which is fricking *spiky* get hurt by doing so

**Comment by Robertha89** [comment]
Yeah, but if you slap people now, it does nothing other then play the sound and create the text, slapping others hurts nobody, unless they are wearing helmets, it makes sense, but it is kinda silly.

Allow us to slap people to death and i see nothing wrong with this.

**Comment by tigercat2000** [comment]
I did say I'm firmly meh about this, but it would probably be better if there was just a custom text instead of damage being done.

**Comment by Robertha89** [comment]
Yes, this is better, the affect of slapping someone is pure fluff, thus slapping someone with a helmet would be the same.

**Comment by Fox-McCloud** [comment]
My concerns are largely on part with @Robertha89's

If this were changed to be fluff, this may gather more support.

**Comment by ZomgPonies** [comment]
Agreed emotes should be purely fluff.

**Comment by Fox-McCloud** [comment]
@FreeStylaLT 

Going to work on this further or...go ahead and close it? It doesn't seem to be gaining much traction =X

**Comment by FreeStylaLT** [comment]
Well, due to me swapping computers recently and losing all of my bash things and having another swap planned soon to a whole new computer again, I'll just go ahead and close this since it will be a while before I will be able to reliably set up the databases again.

Yes, I'm that lazy that I won't set up bash for a deletion of one line. Literally one line.

**Pull request closed without merging**